lib/active_record/virtual_attributes/virtual_arel.rb

```diff
@@ -21,8 +25,11 @@ module ClassMethods
         def arel_attribute(column_name, arel_table = self.arel_table)
           load_schema
           if virtual_attribute?(column_name) && !attribute_alias?(column_name)
-            col = _virtual_arel[column_name.to_s]
-            col.call(arel_table) if col
+            if (col = _virtual_arel[column_name.to_s])
+              arel = col.call(arel_table)
+              arel.name = column_name if arel.kind_of?(Arel::Nodes::Grouping)
```

Comment:
Can we initialize the name when Arel::Nodes::Grouping is created?  Why is this type different?

Reply:
the `call` is calling into every attribute defined.
the name is part of the general arel interface

so, unfortunately, this needs to be here (and is a hack)